docs: improve accessibility with descriptive link and main landmark

MeAkib opened the pull request:
- Replaced <a><button></button></a> with a proper <a> containing text and aria-label to ensure links have discernible text.

- Added a top-level <div role="main" class="docs-app-main-content"> to provide a consistent main landmark, improving accessibility for screen readers and satisfying WCAG and Lighthouse requirements.

Fixes #65181.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/contributing-docs/commit-message-guidelines.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.dev application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: N/A


## What is the new behavior?


## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

JeanMeche commented on adev/src/app/app.component.html:
```diff
@@ -9,7 +9,7 @@
 @if (displaySecondaryNav()) {
   <adev-secondary-navigation />
 }
-<div class="docs-app-main-content">
```
Please see my comment in #65181, we already have the `main` element on most pages.

MeAkib committed:
docs: improve accessibility with descriptive link and main landmark
- Replaced <a><button></button></a> with a proper <a> containing text and
  aria-label to ensure links have discernible text.

- Added main tag in home page to provide
  a consistent main landmark, improving accessibility for screen readers
  and satisfying WCAG and Lighthouse requirements.

Fixes #65181.

thePunderWoman commented:
This PR was merged into the repository. The changes were merged into the following branches:

- main: ee5947db38d56932f6164d45b32e18ca829f4c8f
- 20.3.x: 2e00074c643779e956afae162f5acb5a5d779aa8
- 21.0.x: 70736309f5464f89b274648051f7bd79ae3c3b11
